pojo/src/test/java/com/spotify/hamcrest/pojo/IsPojoTest.java

```diff
@@ -224,16 +228,11 @@ public void testTypeSafeMismatch() throws Exception {
 
   @Test
   public void testMultipleIdenticalMatches() throws Exception {
-    final IsPojo<SomeClass> sut = pojo(SomeClass.class)
+    expectedException.expect(both(isA(IllegalArgumentException.class))
+        .and(hasProperty("message", startsWith("Multiple entries with same key: "))));
```

Comment:
Not sure how to do this change tbh.

 1. I would like to treat this as an "unlikely" user error since it in my opinion is intuitive that you can't register several matchers for the same property (this is in general the behavior of Hamcrest) so I would prefer to not spend too much effort on a good error message here compared to more likely user errors (like "method doesn't exist" or "method threw exception" or "wrong return type" etc)
 2. I would not like to assume that the user used `.where`, but rather say whether `.hasProperty` or `.where` was used if we're going to mention it at all
 3. It would be nice to not couple the error message to something that will eventually become out of date, e.g. `allOf` might be recommended now but maybe not in the future.